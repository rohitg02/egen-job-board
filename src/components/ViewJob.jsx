import React, { useContext, Fragment } from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

import ThemeContext from '../contexts/ThemeContext';

const useStyles = makeStyles({
  root: {
    minWidth: 275,
    height: 250
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  posTop:{
    marginTop: 12,
  },
  onHover : {
    cursor: 'pointer'
  },
  logo: {
    height: '25px',
    width: '50px'
  },
  titleJob: {
    fontSize: '18px',
    fontWeight: 500,
    marginTop: 2,
    marginBottom: 2
  }
});

const ViewJob = ({job, onClickHandler}) => {
  const themeContext= useContext(ThemeContext);
  const classes = useStyles();
  const bull = <span className={classes.bullet}>â€¢</span>;
  const { company_logo, created_at, type, title, company, location } = job;

  return (
    <Card
      id="viewJobCard"
      onClick={() => onClickHandler(job)}
      className={`${classes.onHover} ${themeContext.card.background} ${classes.root}`}
    >
      <CardContent>
        <Fragment>
          <img
           src={company_logo}
           className={classes.logo}
           alt="new"
           />
        </Fragment>
        <Typography className={`${classes.posTop} ${themeContext.card.text}`} variant="body2" component="p" color="textSecondary" >
          {created_at} {bull} {type}
        </Typography>
        <Typography className={`${classes.titleJob} ${themeContext.card.title}`} variant="subtitle1" component="h3">
          {title}
        </Typography>
        <Typography className={`${classes.pos} ${themeContext.card.text}`} variant="body2" component="p" color="textSecondary">
          {company}
        </Typography>
        <Typography variant="body2" component="p">
          <br />
        </Typography>
        <Typography className="cPurple" variant="body2" component="p">
          {location}
        </Typography>
      </CardContent>
    </Card>
  )
}


ViewJob.propTypes = {
  onClickHandler: PropTypes.func,
  job: PropTypes.shape({
    id: PropTypes.string,
    type: PropTypes.string,
    url: PropTypes.string,
    created_at: PropTypes.string,
    company: PropTypes.string,
    company_url: PropTypes.string,
    location: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
    how_to_apply: PropTypes.string,
    company_logo: PropTypes.string
  })
};

ViewJob.defaultProps = {
  onClickHandler: () => {},
  job: {
    id: "",
    type: "",
    url: "",
    created_at: "",
    company: "",
    company_url: "",
    location: "",
    title: "",
    description: "",
    how_to_apply: "",
    company_logo: ""
  }
};

export default ViewJob;
