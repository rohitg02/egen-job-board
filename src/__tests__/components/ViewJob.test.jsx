import React from 'react';
import { shallow } from '../../enzyme';

import ViewJob from '../../components/ViewJob';
import ThemeContext from '../../contexts/ThemeContext';
import light from '../../themes/light';


const mockJob = {
  "id": "b1691d49-ee1d-4793-9c1d-7a10a9b8b84a",
  "type": "Full Time",
  "url": "https://jobs.github.com/positions/b1691d49-ee1d-4793-9c1d-7a10a9b8b84a",
  "created_at": "Mon Dec 07 19:23:19 UTC 2020",
  "company": "EME Hive",
  "company_url": "http://emehive.co",
  "location": "New York City",
  "title": "Infrastructure Engineer for Social Livestreaming App",
  "description": "<p>WHO WE ARE</p>",
  "how_to_apply": "<p>Email your resume and cover letter to</p>",
  "company_logo": "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbFNTIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--f8a7d57b1091ad6c57a6d522d8b280372a303ce8/Screen%20Shot%202020-12-07%20at%202.22.26%20PM.png"
};

describe('ViewJob component', () => {
  it('should render ViewJob component', () => {
    const wrapper = shallow(<ViewJob />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default click handler', () => {
    const wrapper = shallow(<ViewJob />);
    expect(ViewJob.defaultProps.onClickHandler).toBeDefined();
  });

  it('should call onClickHandler', () => {
    const props = {
      job: mockJob,
      onClickHandler: jest.fn((val) => val)
    };
    const wrapper = shallow(<ViewJob {...props} />);
    wrapper.find('#viewJobCard').simulate('click');
    expect(props.onClickHandler).toHaveBeenCalled();
  });
});
