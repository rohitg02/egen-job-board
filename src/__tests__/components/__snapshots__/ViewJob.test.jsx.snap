// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewJob component should render ViewJob component 1`] = `ShallowWrapper {}`;
